File: day12/shared.rb
# frozen_string_literal: true

module Shared

  class ConditionRecord
    attr_reader :springs, :groups

    def initialize(springs, groups)
      @springs = springs
      @groups = groups
    end
  end

  def self.lines_from_file
    file_path = 'data/input.txt'
    IO.readlines(file_path).map { |line| line.chomp.strip }
  end

  def self.condition_records_from_lines(lines)
    lines.map do |line|
      split_line = line.split(' ')
      ConditionRecord.new(split_line[0], split_line[1].split(',').map { |number| Integer(number) })
    end
  end
end

File: day19/shared.py
class Input:
    def __init__(self, workflows, ratings):
        self.workflows = workflows
        self.ratings = ratings


def get_input(file_path):
    with open(file_path) as file:
        workflows = {}
        ratings = []
        blank_line_reached = False
        for line in file.readlines():
            if line == '\n':
                blank_line_reached = True
            elif blank_line_reached:
                ratings.append({split_part.split('=')[0]: int(split_part.split('=')[1]) for split_part in line
                               .strip()[1:-1]
                               .split(',')})
            else:
                split_workflow = line.strip().split('{')
                workflows[split_workflow[0]] = split_workflow[1][0:-1].split(',')
        return Input(workflows, ratings)

